src/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -378,13 +375,20 @@ protected virtual byte[] CustomSerializationData
         /// Allow the association of claims with this instance of <see cref="ClaimsIdentity"/>. 
         /// The claims will not be serialized or added in Clone(). They will be included in searches, finds and returned from the call to <see cref="ClaimsIdentity.Claims"/>.
         /// </summary>               
-        internal Collection<IEnumerable<Claim>> ExternalClaims
+        internal List<List<Claim>> ExternalClaims
         {
-            get { return _externalClaims; }
+            get
+            {
+                if (_externalClaims == null)
+                {
+                    _externalClaims = new List<List<Claim>>();
+                }
+                return _externalClaims;
+            }
```

Comment:
Are we ok with ClaimsIdentity.ExternalClaims not being thread safe? We can lock onto _instanceClaims if needed.